cetmix_tower_server/models/cx_tower_file_template.py

```diff
@@ -53,6 +54,52 @@ def _compute_file_count(self):
         default="tower",
     )
 
+    variable_ids = fields.Many2many(
+        comodel_name="cx.tower.variable",
+        relation="cx_tower_file_template_variable_rel",
+        column1="file_template_id",
+        column2="variable_id",
+        string="Variables",
+        compute="_compute_variable_ids",
+        store=True,
+    )
+
+    @api.depends("code", "server_dir", "file_name")
+    def _compute_variable_ids(self):
+        """
+        Compute and assign the variable_ids field by searching for variables
+        based on references from the code, server_dir and file_name fields.
+        If no variables are found, the field is cleared.
+        """
+        for record in self:
+            all_references = set()
+            if record.code:
+                all_references.update(
+                    self.env["cx.tower.template.mixin"].get_variables_from_code(
+                        record.code
+                    )
+                )
+            if record.server_dir:
+                all_references.update(
+                    self.env["cx.tower.template.mixin"].get_variables_from_code(
+                        record.server_dir
+                    )
+                )
+            if record.file_name:
+                all_references.update(
+                    self.env["cx.tower.template.mixin"].get_variables_from_code(
+                        record.file_name
+                    )
+                )
+
+            if all_references:
+                variables = self.env["cx.tower.variable"].search(
+                    [("reference", "in", list(all_references))]
+                )
+                record.variable_ids = [(6, 0, variables.ids)]
+            else:
+                record.variable_ids = [(5, 0, 0)]
```

Comment:
all `_compute_variable_ids` methods are very similar, we should consider making one common method for such calculations, it will simplify code support.

Reply:
@GabbasovDinar Thanks for the review! I've made changes. Please check and let me know your thoughts.